src/main/java/com/aws/greengrass/logmanager/model/LogFile.java

```diff
@@ -50,29 +51,21 @@ public static LogFile[] of(File... files) {
     }
 
     /**
-     * Read target lines from the file.
-     * The file must contain (minLine + 1) lines. One extra line is needed to prevent incomplete line when hashing.
-     * @return an ArrayList of lines or empty ArrayList
+     * Read target bytes from the file.
+     * @return read byte array.
      */
-    private List<String> getLines() {
-        List<String> linesRead = new ArrayList<>();
-        try (BufferedReader r = Files.newBufferedReader(this.toPath(), StandardCharsets.UTF_8)) {
-            // read target number of lines
-            while (linesRead.size() < linesNeeded) {
-                String oneLine = r.readLine();
-                if (oneLine == null) {
-                    break;
-                }
-                linesRead.add(oneLine);
-            }
+    private byte[] readBytes() {
+        byte[] bytesReadArray = new byte[bytesNeeded];
+        try (InputStream r = Files.newInputStream(this.toPath())) {
+            didRead = r.read(bytesReadArray);
```

Comment:
You should handle `didRead` here, rather than having it as a field of this class. I think that makes the API too complicated.